src/amqp.js

```diff
@@ -0,0 +1,340 @@
+
+import {MarcRecord} from '@natlibfi/marc-record';
+import {Error as ApiError} from '@natlibfi/melinda-commons';
+import createDebugLogger from 'debug';
+import {CHUNK_SIZE} from './constants';
+import httpStatus from 'http-status';
+import {promisify} from 'util';
+
+export async function createAmqpOperator(amqplib, AMQP_URL) {
+  const debug = createDebugLogger('@natlibfi/melinda-record-import-commons:amqp');
+  const debugDev = debug.extend('dev');
+  const debugData = debug.extend('data');
+  const setTimeoutPromise = promisify(setTimeout);
+
+  debug(`Creating an AMQP operator to ${AMQP_URL}`);
+  const connection = await amqplib.connect(AMQP_URL);
+  const channel = await connection.createChannel();
+
+  debug(`Connection: ${connection}`);
+  debug(`Channel: ${channel}`);
+
+  return {purgeQueue, countQueue, getChunk, getOne, ackMessages, nackMessages, sendToQueue, removeQueue, closeChannel, closeConnection};
+
+  async function closeChannel() {
+    debug(`Closing channel`);
+    await channel.close();
+    debug(`Channel: ${channel}`);
+  }
+
+  async function closeConnection() {
+    debug(`Closing connection`);
+    await connection.close();
+    debug(`Connection: ${connection}`);
+  }
+
+  // MARK: Purge queue
+  /**
+   * Purge queue uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @returns {boolean} returns true on succesfull purge
+   */
+  async function purgeQueue({blobId, status}) {
+    const queue = generateQueueId({blobId, status});
```

Comment:
Generating queue id and asserting queue is common for many functions. Maybe introduce a wrapper that does these both (and also validates the queue if it's necessary outside of queueId generation function)?